spec/index.md

```diff
@@ -5033,6 +5104,218 @@ S with
 
 ```
 
+#### IC Management Canister: Take canister snapshot
+
+Only the controllers of the given canister can take a snapshot.
+
+```html
+
+S.messages = Older_messages · CallMessage M · Younger_messages
+(M.queue = Unordered) or (∀ msg ∈ Older_messages. msg.queue ≠ M.queue)
+M.callee = ic_principal
+M.method_name = 'take_canister_snapshot'
+M.arg = candid(A)
+is_system_assigned Snapshot_id    // Snapshot ID is an opaque generated by the system
```

Comment:
The function `is_system_assigned` is defined for principals => let's introduce the variable `Snapshot_id` informally after the sentence "Only the controllers of the given canister can take a snapshot." above.